demos/text_detection_demo/cpp_gapi/src/custom_kernels.cpp

```diff
@@ -0,0 +1,407 @@
+}
+
+GAPI_OCV_KERNEL(OCVCropLabels, custom::CropLabels) {
+    static void run(const cv::Mat& image,
+                    const std::vector<cv::RotatedRect>& rrs,
+                    const std::vector<size_t>& outDims,
+                    const bool centralCrop,
+                          std::vector<cv::Mat>& out,
+                          std::vector<std::vector<cv::Point2f>>& pts) {
+        cv::Size outSize(outDims[3], outDims[2]);
+        std::vector<int> blobShape(4);
+        std::copy(outDims.begin(), outDims.end(), blobShape.begin());
+        const std::vector<cv::Point2f> imgPts { { 0.0f,              0.0f              },
+                                                { image.cols - 1.0f, 0.0f              },
+                                                { image.cols - 1.0f, image.rows - 1.0f },
+                                                { 0.0f,              image.rows - 1.0f } };
+        out.clear();
+        out.reserve(rrs.size());
+        pts.clear();
+        pts.reserve(rrs.size());
+        cv::Mat crop(outSize,   CV_8UC3,  cv::Scalar(0));
+        cv::Mat blob(blobShape, CV_32FC1, cv::Scalar(0));
+        std::vector<cv::Point2f> points(4);
+        for (auto& rr : rrs) {
+            if (rr.size != cv::Size2f(0, 0)) {
+                rr.points(points.data());
+                cropImage(image, points, outSize, crop);
+            } else {
+                if (centralCrop) {
+                    auto r = centralCropRect(image.cols, image.rows);
+                    points = centralCropPoints(r);
+                    cv::resize(image(r), crop, outSize);
+                } else {
+                    points = imgPts;
+                    cv::resize(image, crop, outSize);
+                }
+            }
+            crop.reshape(1, blobShape).convertTo(blob, CV_32F);
+            out.emplace_back(blob.clone());
+            pts.emplace_back(points);
+        }
+    }
+};
+
+GAPI_OCV_KERNEL(OCVCompositeTRDecode, custom::CompositeTRDecode) {
+    static void run(const std::vector<cv::Mat>&  hiddens_,
+                    const std::vector<cv::Mat>&  features,
+                    const cv::gapi::GNetPackage& net,
+                    const size_t                 numClasses,
+                    const size_t                 endToken,
+                          std::vector<cv::Mat>&  res) {
+        constexpr int maxDecodedSymbols = 20;
+        GAPI_DbgAssert(hiddens_.size() == features.size());
+        const size_t numDetectedLabels = features.size();
+
+        std::vector<cv::Mat> states(numDetectedLabels, cv::Mat({1}, CV_32FC1));
+        std::for_each(states.begin(), states.end(),
+                      [](cv::Mat& m){ m.dims = 1; *(m.begin<float>()) = 0; });
+        std::vector<cv::Mat> hiddens(hiddens_);
+
+        res.reserve(numDetectedLabels);
+        for (auto i = 0U; i < numDetectedLabels; i++) {
+            res.emplace_back(cv::Mat({ maxDecodedSymbols, 1, static_cast<int>(numClasses) },
```

Comment:
```suggestion
            res.emplace_back({ maxDecodedSymbols, 1, static_cast<int>(numClasses) },
```

Reply:
No, I need to create `CV_32FC1` Mat

And even without it, vector don't want to create a Mat out of a brace-enclosed initializer list :(

Reply:
Oh, I got what you are saying: `emplace` semantics
Unfortunately, it also can't figure out what is the initializer list. I'll have to add `std::vector<int>`, but I'll fix `emplace` so it will construct Mat in place

Reply:
Done